File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/protocol_generated/NodeTypesParams.kt
@file:Suppress("FunctionName", "ClassName")
package com.sourcegraph.cody.protocol_generated

data class NodeTypesParams(
  var atCursor: String? = null,
  var parent: String? = null,
  var grandparent: String? = null,
  var greatGrandparent: String? = null,
  var lastAncestorOnTheSameLine: String? = null,
)

File: agent/bindings/csharp/NodeTypesParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class NodeTypesParams
  {

    [JsonPropertyName("atCursor")]
    public string AtCursor { get; set; }

    [JsonPropertyName("parent")]
    public string Parent { get; set; }

    [JsonPropertyName("grandparent")]
    public string Grandparent { get; set; }

    [JsonPropertyName("greatGrandparent")]
    public string GreatGrandparent { get; set; }

    [JsonPropertyName("lastAncestorOnTheSameLine")]
    public string LastAncestorOnTheSameLine { get; set; }
  }
}
